scripts/update_from_application_services.py

```diff
@@ -1,185 +1,243 @@
 #!/usr/bin/python3
-
 from pathlib import Path
 from urllib.request import urlopen
 import argparse
 import fileinput
 import hashlib
 import json
-import subprocess
 import shutil
+import subprocess
 import sys
 import tarfile
 import tempfile
 
-ROOT_DIR = Path(__file__).parent.parent
-PACKAGE_SWIFT = ROOT_DIR / "MozillaRustComponents" / "Package.swift"
-# Latest nightly nightly.json file from the taskcluster index
-NIGHTLY_JSON_URL = "https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/project.application-services.v2.nightly.latest/artifacts/public%2Fbuild%2Fnightly.json"
-
-def main():
-    args = parse_args()
-    version = VersionInfo(args.version)
-
-    BRANCH = "rcs-auto-update"
-    TITLE = f"Nightly auto-update ({version.swift_version})"
+ROOT_DIR        = Path(__file__).parent.parent
+PACKAGE_SWIFT   = ROOT_DIR / "MozillaRustComponents" / "Package.swift"
+# Latest nightly.json produced by application‑services CI
+NIGHTLY_JSON_URL = (
+    "https://firefox-ci-tc.services.mozilla.com/api/index/v1/"
+    "task/project.application-services.v2.nightly.latest/artifacts/"
+    "public%2Fbuild%2Fnightly.json"
+)
+
+
+def main() -> None:
+    args     = parse_args()
+    version  = VersionInfo(args.version)
+    BRANCH   = "rcs-auto-update"
+    TITLE    = f"(WIP)(Local AS flow) Nightly auto-update ({version.swift_version})"
+
+    # Ensure we have the latest remote copy of the update branch (if any)
+    subprocess.run(["git", "fetch", args.remote, BRANCH], check=False)
+
+    # Create (or reset) our working branch
+    remote_ref = f"refs/remotes/{args.remote}/{BRANCH}"
+    if subprocess.run(
+        ["git", "rev-parse", "--verify", remote_ref],
+        stdout=subprocess.DEVNULL,
+    ).returncode == 0:
+        subprocess.check_call(["git", "checkout", "-B", BRANCH, remote_ref])
+    else:
+        subprocess.check_call(["git", "checkout", "-B", BRANCH, args.base])
 
+    # Apply the nightly update
     update_source(version)
+
     if not repo_has_changes():
         print("No changes detected, quitting")
-        sys.exit(0)
-    # Make a branch for the PR
-    subprocess.check_call(["git", "checkout", "-B", BRANCH])
-    subprocess.check_call([
-        "git",
-        "commit",
-        "--author",
-        "Firefox Sync Engineering <sync-team@mozilla.com>",
-        "--message",
-        TITLE
-    ])
+        return
+
+    # Stage everything
+    subprocess.check_call(["git", "add", "-A", "MozillaRustComponents"])
```

Comment:
Not sure if that could happen here but when creating automatic PRs I have always tried to specify what to commit, not doing a `git add -A`, with everything changed, sometimes unexpected changes may appear and we want to make sure only those related to the changes are commited.

Reply:
Makes sense I am merging this. @isabelrios let us know if you have any further thoughts on this and we can do a follow-up ( I really want to see this working 😂 )